server/controller/usersController.js

```diff
@@ -0,0 +1,164 @@
+      if (err) {
+        console.error(err);
+        return res.status(StatusCodes.BAD_REQUEST).end();
+      }
+
+      if (results.length > 0) {
+        return res.status(StatusCodes.OK).json(results[0]);
+      } else {
+        return res.status(StatusCodes.NOT_FOUND).json({
+          message: "User not found",
+        });
+      }
+    });
+  });
+};
+
+const join = (req, res) => {
+  const { email, password, location, nickname } = req.body;
+
+  const sql = `INSERT INTO users (email, password, salt, location, nickname) VALUES (?, ?, ?, ?, ?)`;
+
+  const salt = crypto.randomBytes(16).toString("base64");
+  const hashPassword = hashing(password, salt);
+
+  const values = [email, hashPassword, salt, location, nickname];
+  conn.query(sql, values, (err, results) => {
+    if (err) {
+      console.log(err);
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+
+    if (results.affectedRows) {
+      return res.status(StatusCodes.CREATED).json(results);
+    } else {
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+  });
+};
+
+const login = (req, res) => {
+  const { email, password } = req.body;
+
+  const sql = `SELECT * FROM users WHERE email = ?`;
+  conn.query(sql, email, (err, results) => {
+    if (err) {
+      console.log(err);
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+
+    if (results.length === 0) {
+      return res
+        .status(StatusCodes.UNAUTHORIZED)
+        .json({
+          message: "아이디 또는 비밀번호를 잘못 입력했습니다.",
+        })
+        .end();
+    }
+
+    const loginUser = results[0];
+
+    // salt값 꺼내서 날 것으로 들어온 비밀번호 암호화
+    const hashPassword = hashing(password, loginUser.salt);
+
+    // 위에서 암호화된 비밀번호 db 비밀번호랑 비교
+    if (loginUser && loginUser.password == hashPassword) {
+      const access_token = jwt.sign(
+        {
+          id: loginUser.id,
+          email: loginUser.email,
+        },
+        process.env.PRIVATE_KEY,
+        {
+          expiresIn: process.env.TOKEN_EXPIRED_TIME,
+          issuer: "SmartDay",
+        },
+      );
+      // 토큰 쿠키에 담기
+      res.cookie("access_token", access_token, {
+        httpOnly: true,
+      });
+
+      return res.status(StatusCodes.OK).json(results);
+    } else {
+      return res
+        .status(StatusCodes.UNAUTHORIZED)
+        .json({
+          message: "아이디 또는 비밀번호를 잘못 입력했습니다.",
+        })
+        .end();
+    }
+  });
+};
+
+// 회원 정보 수정
+const updateUserInformation = (req, res) => {
+  const { location, nickname } = req.body;
+
+  ensureAuthorization(req, res, () => {
+    const sql = `UPDATE users SET location=?, nickname=? WHERE email=?`;
+
+    const values = [location, nickname, req.authorization.email];
+    conn.query(sql, values, (err, results) => {
+      if (err) {
+        console.error(err);
+        return res.status(StatusCodes.BAD_REQUEST).end();
+      }
+
+      return res.status(StatusCodes.OK).json(results);
+    });
+  });
+};
+
+const hashing = (password, salt) => {
```

Comment:
매개변수로 password를 받고있기 때문에 이렇게 함수명이 표현되는게 더 적절할 것 같아요~
```suggestion
const hashPassword = (password, salt) => {
```